soco/music_services/music_service.py

```diff
@@ -0,0 +1,780 @@
+# -*- coding: utf-8 -*-
+
+"""Sonos Music Services interface.
+
+This module provides the MusicService class and related functionality.
+
+"""
+# pylint: disable=fixme
+
+from __future__ import unicode_literals
+
+import logging
+log = logging.getLogger(__name__)  # pylint: disable=C0103
+
+from xmltodict import parse
+import requests
+
+from soco import SoCo
+from soco.xml import XML
+from soco.exceptions import MusicServiceException
+from soco.soap import SoapMessage, SoapFault
+
+from soco.compat import urlparse, parse_qs
+from soco.music_services.accounts import Account
+
+
+# pylint: disable=too-many-instance-attributes, protected-access
+class MusicServiceSoapClient(object):
+
+    """A SOAP client for accessing Music Services
+
+    This class handles all the necessary authentication for accessing third
+    party music services. You are unlikely to need to use it yourself.
+
+    """
+
+    def __init__(self, endpoint, timeout, music_service):
+        """ Constructor
+
+        Args:
+             endpoint (str): The SOAP endpoint. A url.
+             timeout (int): Timeout the connection after this number of
+                 seconds
+             music_service (MusicService): The MusicService object to which
+                 this client belongs
+
+        """
+
+        self.endpoint = endpoint
+        self.timeout = timeout
+        self.music_service = music_service
+        self.namespace = 'http://www.sonos.com/Services/1.1'
+
+        # Spotify uses gzip. Others may do so as well. Unzipping is handled
+        # for us by the requests library. We need to set the user agent to
+        # a genuine Sonos value. Google Play music seems to want this. The
+        # firmware release number (after 'Sonos/" is obviously fake, and
+        # higher than current values, in case Google only offers certain
+        # services to certain firmware rel eases. ") #
```

Comment:
Ahh. Maybe I remember wrong. I just thought that was one of the info items in the new patch for speaker_info. But you reviewed that, so probably not.